### PR TITLE
partial plots

### PR DESCRIPTION
**Pull request by dchudz**
Empty place-holder PR for partial plots.

Dan, I think you said this (http://scikit-learn.org/stable/auto_examples/ensemble/plot_partial_dependence.html) only works for models in `sklearn.ensemble`? Did it seem like there was any reason for that?

I'm mostly inclined to just start playing around with code to do what we want, but also wanting to know more about why sklearn won't do what we want. Did it seem like there was any fundamental reason for the limitation?

### CODE REVIEW

**Comment by dansbecker** [comment]
@dchudz It is definitely the case that plot_partial_dependence only works for gradient boosting models.  Superficially, you can see that they rely on that assumption in a few places (https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/ensemble/partial_dependence.py#L71) 

At a deeper level, I'm not sure why they've chosen to do it that way.

**Comment by dchudz** [comment]
I've piled on to the issue they have open about it: https://github.com/scikit-learn/scikit-learn/issues/4405

**Comment by dansbecker** [comment]
Cool.  Glad to see you took the initiative, and I think you described the
question perfectly.  Interested to see what the response is.
